app/src/main/java/com/example/drawingApp/DialogUtility.kt

```diff
@@ -0,0 +1,241 @@
+         * ideally setting the state of the boolean to true on the opening of the dialog and false on the closing
+         */
+        onAlertDialogOpenState: (Boolean) -> Unit,
+        /**
+         * passed to the showLayerAlertDialog function on the edit call as its onTextChanged lambda
+         */
+        onAlertDialogTextChange: (String) -> Unit, //handles text change
+        /**
+         * used to assign the index that we are actively editing
+         * Ideally hold an outside variable that we update useing the int with this
+         * pass that variable as the activeEditIndex
+         */
+        onEdit: (Int) -> Unit,
+        /**
+         * used to determine whether we have the sheet open or not
+         * ideally sets a boolean that tracks the open state to false
+         * note: unlike the alert dialog it does not need a onOpen to set true, can simply do that before function call
+         */
+        onDismiss: () -> Unit
+    ): BottomSheetDialog {
+        val data = mutableListOf<LayerViewModel>()
+        for (x in layerNames) data.add(LayerViewModel(x))
+
+        lateinit var myAdapter: EditLayerAdapter
+        myAdapter = EditLayerAdapter(
+            data = data,
+            onDeleteClick = { layerModel ->
+                val index = data.indexOf(layerModel)
+                data.removeAt(index)
+                myAdapter.notifyItemRemoved(index)
+            },
+            onEditClick = { layerModel ->
+                onAlertDialogOpenState(true)
+                val index = data.indexOf(layerModel)
+                onEdit(index)
+                showLayerAlertDialog(
+                    context = context,
+                    layerNames = layerNames,
+                    refText = alertDialogRefText,
+                    wasOpen = wasAlertOpen,
+                    isCreating = false,
+                    onTextChanged = onAlertDialogTextChange,
+                    onDialogSubmission = {
+                        it?.let {
+                            data[index] = LayerViewModel(it)
+                            layerNames[index] = it
+                            myAdapter.notifyItemChanged(index)
+                        }
+                        onAlertDialogOpenState(false)
+                    },
+                    onOpen = { onAlertDialogOpenState(true) }
+                )
+            }
+        )
+
+        val sheet = BottomSheetDialog(context)
+        val inflater = LayoutInflater.from(context)
+        val view = inflater.inflate(R.layout.edit_layer_recycler, null)
+        view.findViewById<RecyclerView>(R.id.editLayerRecycler).apply {
+            layoutManager = LinearLayoutManager(context)
+            adapter = myAdapter
+        }
+
+        sheet.setContentView(view)
+        sheet.setOnDismissListener { onDismiss() }
+        sheet.show()
+
+        if (wasAlertOpen && activeEditIndex != -1) {
```

Comment:
the activeEditIndex at -1 here strikes me as odd. I feel like you basically want a nullable here but are doing it in a different way?

yeah after looking at all the points around the code referencing this variable, I think it should be a nullable, this one does feel wrong